<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>2200</width>
    <height>1500</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>Алгоритм отсечения отрезка произвольным выпуклым отсекателем (Алгоритм Кируса—Бека)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>200</y>
      <width>441</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Цвет отрезков</string>
    </property>
    <widget class="QComboBox" name="box_color_segment">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>391</width>
       <height>71</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Черный</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Синий</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Зеленый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Красный</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Розовый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Бирюзовый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Голубой</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Желтый</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGraphicsView" name="graph">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>30</y>
      <width>1600</width>
      <height>1100</height>
     </rect>
    </property>
   </widget>
   <widget class="QTableWidget" name="table_segments">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>1190</y>
      <width>791</width>
      <height>151</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>7</pointsize>
     </font>
    </property>
    <property name="columnCount">
     <number>4</number>
    </property>
    <column>
     <property name="text">
      <string>X1</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Y1</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>X2</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Y2</string>
     </property>
    </column>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>1150</y>
      <width>841</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Введенные отрезки</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>30</y>
      <width>441</width>
      <height>161</height>
     </rect>
    </property>
    <property name="title">
     <string>Цвет отсекателя</string>
    </property>
    <widget class="QComboBox" name="box_color_cutter">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>391</width>
       <height>71</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Красный</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Черный</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Зеленый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Розовый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Бирюзовый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Голубой</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Синий</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Желтый</string>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>380</y>
      <width>441</width>
      <height>241</height>
     </rect>
    </property>
    <property name="title">
     <string>Цвет и толщина результата</string>
    </property>
    <widget class="QComboBox" name="box_color_result">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>391</width>
       <height>71</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>Зеленый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Черный</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Розовый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Бирюзовый</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Голубой</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Красный</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Синий</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Желтый</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>160</y>
       <width>181</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Толщина:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="box_width">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>160</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="but_connect">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>950</y>
      <width>441</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Замнкнуть отсекатель</string>
    </property>
   </widget>
   <widget class="QPushButton" name="but_cut">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>1160</y>
      <width>441</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Отсечь</string>
    </property>
   </widget>
   <widget class="QPushButton" name="but_clean">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>1240</y>
      <width>441</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить экран</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>1360</y>
      <width>2161</width>
      <height>111</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Левая кнопка мыши - ввод вершины отрезка; правая - ввод вершины отсекателя.</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>720</y>
      <width>441</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string>Ввод вершины</string>
    </property>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>47</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>X:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>60</y>
       <width>47</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Y:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="box_x">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>60</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="singleStep">
      <number>100</number>
     </property>
     <property name="value">
      <number>200</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="box_y">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>60</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10000</number>
     </property>
     <property name="maximum">
      <number>10000</number>
     </property>
     <property name="singleStep">
      <number>100</number>
     </property>
     <property name="value">
      <number>200</number>
     </property>
    </widget>
    <widget class="QPushButton" name="but_input_point_segment">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>140</y>
       <width>201</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Отрезок</string>
     </property>
    </widget>
    <widget class="QPushButton" name="but_input_point_cutter">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>201</width>
       <height>51</height>
      </rect>
     </property>
     <property name="text">
      <string>Отсекатель</string>
     </property>
    </widget>
   </widget>
   <widget class="QTableWidget" name="table_cutter">
    <property name="geometry">
     <rect>
      <x>1440</x>
      <y>1190</y>
      <width>471</width>
      <height>151</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>7</pointsize>
     </font>
    </property>
    <property name="columnCount">
     <number>2</number>
    </property>
    <column>
     <property name="text">
      <string>X1</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Y1</string>
     </property>
    </column>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>1420</x>
      <y>1150</y>
      <width>521</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Введенные вершины отсекателя</string>
    </property>
   </widget>
   <zorder>groupBox_6</zorder>
   <zorder>groupBox_4</zorder>
   <zorder>groupBox</zorder>
   <zorder>graph</zorder>
   <zorder>table_segments</zorder>
   <zorder>groupBox_5</zorder>
   <zorder>groupBox_2</zorder>
   <zorder>but_connect</zorder>
   <zorder>but_cut</zorder>
   <zorder>but_clean</zorder>
   <zorder>label</zorder>
   <zorder>groupBox_3</zorder>
   <zorder>table_cutter</zorder>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
